### PR TITLE
Cut v0.5 Release

### PR DESCRIPTION
**Pull request by asifdxtreme**
Cut v0.5 Release and ChangeLog for new release

### CODE REVIEW

**Comment by asifdxtreme** [comment]
@k82cn I have made the kube-batch image for v0.5 and pushed over [here](https://hub.docker.com/r/kubesigs/kube-batch/tags) and tested with basic [example.](https://github.com/kubernetes-sigs/kube-batch/blob/master/example/job.yaml)

File: CHANGLOG.md [hunk 1]
```diff
@@ -1,3 +1,164 @@
...
+- Fix e2e test: gang scheduling. [\#835](https://github.com/kubernetes-sigs/kube-batch/issues/835)
+- Update API GroupName [\#815](https://github.com/kubernetes-sigs/kube-batch/issues/815)
+- Migrate nodeorder and predicates plugins [\#814](https://github.com/kubernetes-sigs/kube-batch/issues/814)
+- some files are repeated in multiple folders [\#804](https://github.com/kubernetes-sigs/kube-batch/issues/804)
+- the job cannot be scheduled [\#803](https://github.com/kubernetes-sigs/kube-batch/issues/803)
+- Add Configuration for predicate plugin to enable/disable predicates algorithm [\#802](https://github.com/kubernetes-sigs/kube-batch/issues/802)
+- Propose not to delete all pods after the job finished [\#797](https://github.com/kubernetes-sigs/kube-batch/issues/797)
+- verify-gencode.sh can not work [\#788](https://github.com/kubernetes-sigs/kube-batch/issues/788)
+- Consider support multi-containers pod error code handling [\#776](https://github.com/kubernetes-sigs/kube-batch/issues/776)
+- vkctrl enhancement [\#774](https://github.com/kubernetes-sigs/kube-batch/issues/774)
```
Comment (k82cn) [inline CHANGLOG.md]: why do we include this issue?

File: CHANGLOG.md [hunk 2]
```diff
@@ -1,3 +1,164 @@
...
+- Fix e2e test: statement [\#836](https://github.com/kubernetes-sigs/kube-batch/issues/836)
+- Fix e2e test: gang scheduling. [\#835](https://github.com/kubernetes-sigs/kube-batch/issues/835)
+- Update API GroupName [\#815](https://github.com/kubernetes-sigs/kube-batch/issues/815)
+- Migrate nodeorder and predicates plugins [\#814](https://github.com/kubernetes-sigs/kube-batch/issues/814)
+- some files are repeated in multiple folders [\#804](https://github.com/kubernetes-sigs/kube-batch/issues/804)
+- the job cannot be scheduled [\#803](https://github.com/kubernetes-sigs/kube-batch/issues/803)
+- Add Configuration for predicate plugin to enable/disable predicates algorithm [\#802](https://github.com/kubernetes-sigs/kube-batch/issues/802)
+- Propose not to delete all pods after the job finished [\#797](https://github.com/kubernetes-sigs/kube-batch/issues/797)
+- verify-gencode.sh can not work [\#788](https://github.com/kubernetes-sigs/kube-batch/issues/788)
+- Consider support multi-containers pod error code handling [\#776](https://github.com/kubernetes-sigs/kube-batch/issues/776)
```
Comment (k82cn) [inline CHANGLOG.md:14]: ditto

**Comment by k82cn** [comment]
please only valid issue in changelogs.

File: CHANGLOG.md (hunk 1 above)
Comment (asifdxtreme) [reply CHANGLOG.md]: updated

File: CHANGLOG.md (hunk 2 above)
Comment (asifdxtreme) [reply CHANGLOG.md:14]: updated

**Comment by asifdxtreme** [comment]
> please only valid issue in changelogs.

Removed all the trivial issues from the list

File: CHANGLOG.md [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+- Helm install fails [\#857](https://github.com/kubernetes-sigs/kube-batch/issues/857)
+- "NewResource/Convert2K8sResource" behavior mismatch [\#851](https://github.com/kubernetes-sigs/kube-batch/issues/851)
+- Phase of podgroup status looks incorrect [\#846](https://github.com/kubernetes-sigs/kube-batch/issues/846)
+- Update API GroupName [\#815](https://github.com/kubernetes-sigs/kube-batch/issues/815)
+- Migrate nodeorder and predicates plugins [\#814](https://github.com/kubernetes-sigs/kube-batch/issues/814)
+- Add Configuration for predicate plugin to enable/disable predicates algorithm [\#802](https://github.com/kubernetes-sigs/kube-batch/issues/802)
+- Propose not to delete all pods after the job finished [\#797](https://github.com/kubernetes-sigs/kube-batch/issues/797)
+- Consider support multi-containers pod error code handling [\#776](https://github.com/kubernetes-sigs/kube-batch/issues/776)
+- Support Task/Job retry [\#769](https://github.com/kubernetes-sigs/kube-batch/issues/769)
+- Job GC [\#768](https://github.com/kubernetes-sigs/kube-batch/issues/768)
```
Comment (k82cn) [inline CHANGLOG.md]: we already move volcano out, why include this issue?

File: CHANGLOG.md [hunk 4]
```diff
@@ -1,3 +1,37 @@
...
+- Deserved attr is not correctly calculated in proportion plugin [\#729](https://github.com/kubernetes-sigs/kube-batch/issues/729)
+- Keep backward compatibility for priority class [\#724](https://github.com/kubernetes-sigs/kube-batch/issues/724)
+- Change return value of NodeOrderFn from int to float [\#708](https://github.com/kubernetes-sigs/kube-batch/issues/708)
+- Add type Argument with some common parse function [\#704](https://github.com/kubernetes-sigs/kube-batch/issues/704)
+- Replace NodeOrder with BestNode [\#699](https://github.com/kubernetes-sigs/kube-batch/issues/699)
+- Support set default value to the configuration [\#695](https://github.com/kubernetes-sigs/kube-batch/issues/695)
+- Add resource predicates for tasks [\#694](https://github.com/kubernetes-sigs/kube-batch/issues/694)
+- Pass conformance test [\#589](https://github.com/kubernetes-sigs/kube-batch/issues/589)
+- big PodGroup blocks scheduling issue [\#514](https://github.com/kubernetes-sigs/kube-batch/issues/514)
+- Changing the architecture diagram [\#504](https://github.com/kubernetes-sigs/kube-batch/issues/504)
```
Comment (k82cn) [inline CHANGLOG.md]: what's updated in this issue?

File: CHANGLOG.md [hunk 5]
```diff
@@ -1,3 +1,37 @@
...
+- Support MPI job by kube-batch [\#351](https://github.com/kubernetes-sigs/kube-batch/issues/351)
+- Currently no IBM Power\(ppc64le\) based image is available in kubearbitrator public repository [\#241](https://github.com/kubernetes-sigs/kube-batch/issues/241)
```
Comment (k82cn) [inline CHANGLOG.md]: ditto

File: CHANGLOG.md [hunk 6]
```diff
@@ -1,3 +1,37 @@
...
+- Keep backward compatibility for priority class [\#724](https://github.com/kubernetes-sigs/kube-batch/issues/724)
+- Change return value of NodeOrderFn from int to float [\#708](https://github.com/kubernetes-sigs/kube-batch/issues/708)
+- Add type Argument with some common parse function [\#704](https://github.com/kubernetes-sigs/kube-batch/issues/704)
+- Replace NodeOrder with BestNode [\#699](https://github.com/kubernetes-sigs/kube-batch/issues/699)
+- Support set default value to the configuration [\#695](https://github.com/kubernetes-sigs/kube-batch/issues/695)
+- Add resource predicates for tasks [\#694](https://github.com/kubernetes-sigs/kube-batch/issues/694)
+- Pass conformance test [\#589](https://github.com/kubernetes-sigs/kube-batch/issues/589)
+- big PodGroup blocks scheduling issue [\#514](https://github.com/kubernetes-sigs/kube-batch/issues/514)
+- Changing the architecture diagram [\#504](https://github.com/kubernetes-sigs/kube-batch/issues/504)
+- Support MPI job by kube-batch [\#351](https://github.com/kubernetes-sigs/kube-batch/issues/351)
```
Comment (k82cn) [inline CHANGLOG.md]: ditto

**Comment by asifdxtreme** [comment]
@k82cn i have updated the change log as per your feedback, these changelogs are generated automatically based on the issues closed between the release tags, probably we can add a new label(`release-notes`) to the issues which needs to be mentioned in the ReleaseNotes/Changelog

File: CHANGLOG.md (hunk 2 above)
Comment (k82cn) [reply CHANGLOG.md:14]: still there?

File: CHANGLOG.md [hunk 7]
```diff
@@ -1,3 +1,32 @@
...
+- master and worker start in order in a job [\#872](https://github.com/kubernetes-sigs/kube-batch/issues/872)
+-  use kubernetes default schedule policies [\#864](https://github.com/kubernetes-sigs/kube-batch/issues/864)
+- v0.4 runtime panic  [\#861](https://github.com/kubernetes-sigs/kube-batch/issues/861)
+- Helm install fails [\#857](https://github.com/kubernetes-sigs/kube-batch/issues/857)
+- "NewResource/Convert2K8sResource" behavior mismatch [\#851](https://github.com/kubernetes-sigs/kube-batch/issues/851)
+- Phase of podgroup status looks incorrect [\#846](https://github.com/kubernetes-sigs/kube-batch/issues/846)
+- Update API GroupName [\#815](https://github.com/kubernetes-sigs/kube-batch/issues/815)
+- Migrate nodeorder and predicates plugins [\#814](https://github.com/kubernetes-sigs/kube-batch/issues/814)
+- Add Configuration for predicate plugin to enable/disable predicates algorithm [\#802](https://github.com/kubernetes-sigs/kube-batch/issues/802)
+- Propose not to delete all pods after the job finished [\#797](https://github.com/kubernetes-sigs/kube-batch/issues/797)
```
Comment (k82cn) [inline CHANGLOG.md]: this is volcano feature.

File: CHANGLOG.md [hunk 8]
```diff
@@ -1,3 +1,32 @@
...
+- Consider support multi-containers pod error code handling [\#776](https://github.com/kubernetes-sigs/kube-batch/issues/776)
+- Support Task/Job retry [\#769](https://github.com/kubernetes-sigs/kube-batch/issues/769)
```
Comment (k82cn) [inline CHANGLOG.md]: ditto

File: CHANGLOG.md [hunk 9]
```diff
@@ -1,3 +1,32 @@
...
+- Support PriorityClassName in Job Resource [\#766](https://github.com/kubernetes-sigs/kube-batch/issues/766)
```
Comment (k82cn) [inline CHANGLOG.md]: ditto

File: CHANGLOG.md [hunk 10]
```diff
@@ -1,3 +1,32 @@
...
+- Support ScheduledJob/CronJob  [\#765](https://github.com/kubernetes-sigs/kube-batch/issues/765)
```
Comment (k82cn) [inline CHANGLOG.md]: ditto

File: CHANGLOG.md [hunk 11]
```diff
@@ -1,3 +1,32 @@
...
+- Queue controller & cli [\#763](https://github.com/kubernetes-sigs/kube-batch/issues/763)
```
Comment (k82cn) [inline CHANGLOG.md]: ditto

**Comment by k82cn** [comment]
> i  have updated the change log as per your feedback, 

:(, I have to add comments one by one .

> , probably we can add a new label(release-notes) to the issues which needs to be mentioned in the ReleaseNotes/Changelog

That's ok to me.

File: CHANGLOG.md (hunk 2 above)
Comment (k82cn) [reply CHANGLOG.md:14]: still there?

**Pull request merged**